test/test_nn.py

```diff
@@ -1556,6 +1556,90 @@ def test_module_to_argparse(self):
         with self.assertRaises(TypeError):
             net.to(cpu, torch.tensor(3, dtype=torch.long), non_blocking=True)
 
+    @unittest.skipIf(not TEST_CUDA, 'CUDA not available')
+    def test_global_flag_overwrite_module_params_on_conversion(self):
+        torch.__future__.set_overwrite_module_params_on_conversion(False)
+
+        # yf225 TODO: add comment for this test! Saying that the current default behavior is just wrong
+        m = nn.Linear(2,3)
+        mw = m.weight[:]
+        m.to('cuda')
+        mw[0][0] = 5
+        with self.assertRaisesRegex(RuntimeError, "Expected object of backend CPU but got backend CUDA"):
```

Comment:
this still looks strange, how does it make sense to have an assert inside an assert?  Just run the code inside the nested assert?

Reply:
Fixed.